bad-code/backend/src/aws.ts

```diff
@@ -49,20 +56,21 @@ export async function copyS3Folder(sourcePrefix: string, destinationPrefix: stri
         if (!listedObjects.Contents || listedObjects.Contents.length === 0) return;
         
         // Copy each object to the new location
-        // Bounty $25 to make this function run parallelly
-        for (const object of listedObjects.Contents) {
-            if (!object.Key) continue;
+        await Promise.all(listedObjects.Contents.map(async (object) => {
+            if (!object.Key) return;
             let destinationKey = object.Key.replace(sourcePrefix, destinationPrefix);
             let copyParams = {
                 Bucket: process.env.S3_BUCKET ?? "",
                 CopySource: `${process.env.S3_BUCKET}/${object.Key}`,
                 Key: destinationKey
             };
-            console.log(copyParams)
 
-            await s3.copyObject(copyParams).promise();
-            console.log(`Copied ${object.Key} to ${destinationKey}`);
-        }
+            console.log(copyParams);
+
+            return s3.copyObject(copyParams).promise().then(() => {
+                console.log(`Copied ${object.Key} to ${destinationKey}`);
+            });
```

Comment:
Hey, since async/await is being used everywhere else, I think you should also use await here. You can revert it to how it was before. Thanks!

Reply:
sure